google/cloud/bigquery/dbapi/_helpers.py

```diff
@@ -113,6 +114,135 @@ def array_to_query_parameter(value, name=None, query_parameter_type=None):
     return bigquery.ArrayQueryParameter(name, array_type, value)
 
 
+complex_query_parameter_parse = re.compile(
+    r"""
+    \s*
+    (ARRAY|STRUCT|RECORD)  # Type
+    \s*
+    <([A-Z0-9<> ,]+)>      # Subtype(s)
+    \s*$
+    """,
+    re.IGNORECASE | re.VERBOSE,
+    ).match
+parse_struct_field = re.compile(
+    r"""
+    (?:(\w+)\s+)    # field name
+    ([A-Z0-9<> ,]+)  # Field type
```

Comment:
This also matches invalid struct fields such as `name str<>ing` - is that fine and we rely on the higher-level logic to detect and handle these cases?

Reply:
Yup. This is just used to pull out the name. The type is then parsed further by calling the same function recursively.